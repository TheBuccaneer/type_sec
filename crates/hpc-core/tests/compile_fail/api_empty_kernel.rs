use hpc_core::*;

/*
trying to invoke empty buffer without data
 */


fn main() {
    let ctx = Context::create_context().unwrap();
    let queue = ctx.create_queue().unwrap();


    let buf = ctx.create_empty_buffer::<u8>(16).unwrap();

    let kernel = Kernel::from_source(&ctx, "kernel void dummy(__global uchar* buf) {}", "dummy").unwrap();
    let (_inflight, _evt) = buf.enqueue_kernel(&queue, &kernel, 16).unwrap();
}
