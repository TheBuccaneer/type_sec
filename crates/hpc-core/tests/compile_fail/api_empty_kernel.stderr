error[E0599]: no method named `enqueue_kernel` found for struct `DeviceBuffer<'_, u8, hpc_core::Empty>` in the current scope
  --> tests/compile_fail/api_empty_kernel.rs:16:33
   |
16 |     let (_inflight, _evt) = buf.enqueue_kernel(&queue, &kernel, 16).unwrap();
   |                                 ^^^^^^^^^^^^^^ method not found in `DeviceBuffer<'_, u8, hpc_core::Empty>`
   |
   = note: the method was found for
           - `DeviceBuffer<'brand, T, Written>`
